File: Src/java/cqf-fhir-npm/src/main/java/org/cqframework/fhir/npm/ILibraryReader.java
package org.cqframework.fhir.npm;

import java.io.IOException;
import java.io.InputStream;
import org.hl7.fhir.exceptions.FHIRFormatError;
import org.hl7.fhir.r5.model.Library;

public interface ILibraryReader {
    public Library readLibrary(InputStream stream) throws FHIRFormatError, IOException;
}

File: cqf-fhir-npm/src/main/kotlin/org/cqframework/fhir/npm/ILibraryReader.kt
package org.cqframework.fhir.npm

import java.io.IOException
import java.io.InputStream
import org.hl7.fhir.exceptions.FHIRFormatError
import org.hl7.fhir.r5.model.Library

interface ILibraryReader {
    @Throws(FHIRFormatError::class, IOException::class)
    fun readLibrary(stream: InputStream?): Library?
}
